packages/static-wado-creator/lib/dictionary/Tags.js

```diff
@@ -9,21 +37,65 @@ const Tags = {
   // This one isn't defined in the dataDictionary
   AvailableTransferSyntaxUID: "00083002",
 
-  // TODO - make this actually work as a real deduplicated object with different creators/lookups.
-  DeduppedCreator: "dedupped",
-  // Creator tag value
-  DeduppedTag: "00090010",
+  DeduppedCreator,
 
   // The references to extract data included in this object, 1..n values
-  DeduppedRef: "00091010",
+  DeduppedRef: { creator: DeduppedCreator, tag: "00091010"},
 
   // The hash value of THIS object
-  DeduppedHash: "00091011",
+  DeduppedHash: {creator: DeduppedCreator, tag: "00091011"},
 
   // Type of hash instance
   DeduppedType: "00091012",
   InstanceType: "instance",
   InfoType: "info",
+
+  setValue: (item, tag, value) => {
+    const actualTag = findPrivate(item,tag,true);
+    item[actualTag] = { Value: Array.isArray(value) ? value : [value] };
+    return actualTag;
+  },
+
+  setList: (item, tag, value) => {
+    const actualTag = findPrivate(item,tag,true);
+    item[actualTag] =  {Value: Array.isArray(value) ? value : [value]};
+    return actualTag;
+  },
+
+  pushList: (item, tag, value) => {
+    const actualTag = findPrivate(item,tag,true);
+    if( !item[actualTag] ) item[actualTag] = { Value: [] };
+    item[actualTag].Value.push(value);
+    return actualTag;
+  },
+
+  getValue: (item, tag) => {
+    const actualTag = findPrivate(item,tag);
+    if( !actualTag ) return undefined;
+    const value = item[actualTag];
+    if( value===undefined ) return undefined;
+    if( Array.isArray(value) ) return value.length==1 ? value[0] : (value.length===0 ? "" : value);
+    if( value.Value ) return value.Value.length==1 ? value.Value[0] : (value.Value.length===0 ? "" : value.Value);
```

Comment:
Will value ever not be {Value: [...]} here? I guess maybe in the case of data that was generated by previous code.

Reply:
I'm starting to set things up to map to the normalized dcmjs format, as that is quite a bit easier and faster to work with, and for loading large studies it makes quite a difference.